tree/ntuple/v7/src/RColumn.cxx

```diff
@@ -61,10 +63,25 @@ void ROOT::Experimental::Detail::RColumn::Connect(DescriptorId_t fieldId, RPageS
 
 void ROOT::Experimental::Detail::RColumn::Flush()
 {
-   if (fHeadPage.GetSize() == 0) return;
+   auto otherIdx = (fHeadPageIdx + 1) % 2;
+   if (fHeadPage[fHeadPageIdx].GetSize() == 0 && fHeadPage[otherIdx].GetSize() == 0)
+      return;
 
-   fPageSink->CommitPage(fHandleSink, fHeadPage);
-   fHeadPage.Reset(fNElements);
+   if ((fHeadPage[fHeadPageIdx].GetNElements() < fApproxNElementsPerPage / 2) &&
+       fHeadPage[otherIdx].GetNElements())
+   {
+      // Small tail page: merge with previously used page
+      void *dst = fHeadPage[otherIdx].TryGrow(fHeadPage[fHeadPageIdx].GetNElements());
+      R__ASSERT(dst != nullptr);
+      RColumnElementBase elem(fHeadPage[fHeadPageIdx].GetBuffer(), fHeadPage[fHeadPageIdx].GetElementSize());
+      elem.WriteTo(dst, fHeadPage[fHeadPageIdx].GetNElements());
+      fHeadPage[fHeadPageIdx].Reset(0);
+      fHeadPageIdx = otherIdx;
```

Comment:
Haven't I seen this code before 😅 Please consider consolidating it into one single function. If there's a bug we will need to remember to change it at three places, at least for me that's unlikely to happen.